src/java.base/share/classes/java/util/Random.java

```diff
@@ -149,6 +285,18 @@ public Random(long seed) {
     private static long initialScramble(long seed) {
         return (seed ^ multiplier) & mask;
     }
+    /**
+     * Returns an instance of {@link Random} based on this
+     * {@code java.util.random.RandomGenerator}. If this generator is already an instance of
+     * {@code Random}, it is returned. Otherwise, this method returns an instance of
+     * {@code Random} that delegates all methods except setSeed to this generator.
+     * Its setSeed method always throws {@link UnsupportedOperationException}.
+     * @param random the {@link java.util.random.RandomGenerator} to use
+     * @return {@link Random}
+     */
+    public static Random from(RandomGenerator random) {
+        return RandomWrapper.wrap(random);
```

Comment:
Might be good to check here or in the called methods / constructors for `null`. Currently `null` would not be noticed until the first method is called on the created `Random`, which makes it difficult for the user to track down bugs in their code.

Reply:
```suggestion
        Objects.requireNonNull(random);
        return RandomWrapper.wrap(random);
```
fyi this is the original change suggested by marcono1234. Notice that this might need a csr update; maybe a javadoc update is needed too